scripts/generate-docs.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+# This script will generate documentation for the current version
+
+VERSION=`python3 -c "import json; f = open('./package.json', 'r'); v = json.loads(f.read())['version']; f.close(); print(v)"`
+
+# Ensure we fail if any of these commands fail
+set -e
+
+yarn gendoc
+
+git checkout gh-pages
+mv .typedoc/ $VERSION/
+sed "6i\    <li><a href=\"$VERSION/index.html\">$VERSION</a></li>" index.html > index.html.new
+rm index.html
+mv index.html.new index.html
+
+git add "$VERSION/"
+git add index.html
+git commit -m "Add documentation for $VERSION"
+git push
```

Comment:
When using automated commits and pushs some checks should have been made that the preceding commands did the expected thing (e.g. not fail or actually create the files/folders).

Reply:
Should be covered by line 6? It will fail the script if any of these return non zero.